### PR TITLE
fix `resolveConfig` to work with Prettier 3

### PR DESCRIPTION
**Pull request by jkrehm**
The `sync` function is gone and only async version exists.

Also remove `eslintPath` argument that test function does not consume.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ async function format(options) {
     // Let prettier infer the parser using the filepath, if present. Otherwise
     // assume the file is JS and default to the babel parser.
     filePath ? { filepath: filePath } : { parser: 'babel' },
-    getPrettierConfig(filePath, prettierPath),
+    await getPrettierConfig(filePath, prettierPath) || {},
```
Comment (JounQin) [inline src/index.js:68]: Why you refactor to move `|| {}` here?

Comment (jkrehm) [reply src/index.js:68]: So we don't have to await in `getPrettierConfig`. As I look at it, a ternary in `getPrettierConfig` might be even better.

```
return prettier.resolveConfig
  ? prettier.resolveConfig(filePath)
  : {};
```

What do you think?

Comment (JounQin) [reply src/index.js:68]: That makes more sense to me.

Comment (JounQin) [reply src/index.js:68]:
```suggestion
    await getPrettierConfig(filePath, prettierPath),
```

File: src/index.js [hunk 2]
```diff
@@ -298,12 +298,8 @@ async function getESLintConfig(filePath, eslintPath, eslintOptions) {
 
 function getPrettierConfig(filePath, prettierPath) {
   const prettier = requireModule(prettierPath, 'prettier');
-  return (
-    (prettier.resolveConfig &&
-      prettier.resolveConfig.sync &&
-      prettier.resolveConfig.sync(filePath)) ||
-    {}
-  );
+
```
Comment (JounQin) [inline src/index.js:301]:
```suggestion
```

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
@jkrehm Thanks for your contribution! This PR has been released at [v16.1.2](https://github.com/prettier/prettier-eslint/releases/tag/v16.1.2)

**Pull request merged**